lambda/family_separation.py

```diff
@@ -0,0 +1,198 @@
+from datetime import datetime, timedelta
+import io
+import json
+import logging
+import os
+import traceback
+from typing import Dict
+
+import boto3
+import requests
+from apiclient.discovery import build
+from mapbox import Geocoder
+from PIL import Image
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+def read_sheet(sheet_range, fields, location_idx, affiliate):
+    log.info('\nload sheet %s with %s', os.environ['SHEET_ID'],
+             os.environ['GOOGLE_API_KEY'])
+    service = build('sheets', 'v4', developerKey=os.environ['GOOGLE_API_KEY'], cache_discovery=False)
+    result = service.spreadsheets().values().get(
+        spreadsheetId=os.environ['SHEET_ID'], range=sheet_range).execute()
+    values = result.get('values', [])
+    rows = {}
+    empty = {}
+    for field in fields:
+        empty[field] = ''
+    idx = 0
+    for row in values[1:]:
+        if not isinstance(location_idx, list):
+            location_idx = [location_idx]
+
+        if max(location_idx) >= len(row):
+            log.warning('Skipping row %d because no location', idx)
+            continue
+
+        props = {}
+        props.update(empty)
+        for field in fields:
+            idx = fields[field]
+            if idx < len(row):
+                val = row[idx].strip()
+                if val:
+                    props[field] = val
+
+                # Y|N to boolean
+                if props[field] == 'Y':
+                    props[field] = True
+                if props[field] == 'N':
+                    props[field] = False
+        location = ' '.join(map(lambda f: row[f].strip(), location_idx))
+        if location:
+            rows[location] = {
+                'id': location,
+                'properties': props
+            }
+            log.debug('row %s\t%s\t%s',
+                      len(rows) + 1, props['name'], rows[location]['id'])
+        else:
+            log.warning('Skipping "%s" at row %s: no location', props['name'], idx)
+        idx += 1
+    log.info('read %s rows from sheet', len(rows))
+    return rows
+
+
+def get_event_data():
+    fields = {
+        'name': 0,
+        'eventDate': 1,
+        'eventLink': 2,
+        'city': 3,
+        'state': 4,
+        'country': 5,
+        'host': 6,
+        'affiliate': 7,
+        'contactName': 8,
+        'contactEmail': 9,
+        'facebook': 10,
+        'twitter': 11,
+        'instagram': 12,
+    }
+    sheet = read_sheet('Sheet1!A1:M', fields, [3, 4, 5], False)
+
+    # add default name
+    for loc in sheet:
+        if not sheet[loc]['properties']['name']:
+            sheet[loc]['properties']['name'] = '%s Event' % loc
```

Comment:
'loc' in this case is the key, which might look very strange. Maybe use the city instead.